pkg/auth/dependency/interaction/provider.go

```diff
@@ -12,14 +12,24 @@ import (
 type IdentityProvider interface {
 	Get(userID string, typ authn.IdentityType, id string) (*IdentityInfo, error)
 	GetByClaims(typ authn.IdentityType, claims map[string]interface{}) (string, *IdentityInfo, error)
+	ListByClaims(claims map[string]string) ([]*IdentityInfo, error)
```

Comment:
`GetByClaims ` and `ListByClaims` treat the claims differently, better to add some comments here?